src/log_surgeon/finite_automata/DfaState.hpp

```diff
@@ -59,20 +82,62 @@ class DfaState {
 template <StateType state_type>
 auto DfaState<state_type>::get_dest_state(uint32_t character) const -> DfaState const* {
     if constexpr (StateType::Byte == state_type) {
-        return m_bytes_transition[character];
+        return m_bytes_transition[character].get_dest_state();
     } else {
         if (character < cSizeOfByte) {
-            return m_bytes_transition[character];
+            return m_bytes_transition[character].get_dest_state();
         }
         std::unique_ptr<std::vector<typename Tree::Data>> result
                 = m_tree_transitions.find(Interval(character, character));
         assert(result->size() <= 1);
         if (false == result->empty()) {
-            return result->front().m_value;
+            return result->front().m_value.get_dest_state();
         }
         return nullptr;
     }
 }
+
+template <StateType state_type>
+auto DfaState<state_type>::serialize(std::unordered_map<DfaState const*, uint32_t> const& state_ids
+) const -> std::string {
+    auto const accepting_tags_string = is_accepting()
+                                               ? fmt::format(
+                                                         "accepting_tags={{{}}},",
+                                                         fmt::join(m_matching_variable_ids, ",")
+                                                 )
+                                               : "";
```

Comment:
```suggestion
    auto const accepting_tags_string{
            is_accepting()
                    ? fmt::format("accepting_tags={{{}}},", fmt::join(m_matching_variable_ids, ","))
                    : ""
    };
```
Maybe use `{}` initialization? Even though it's not enforced for `auto`, it will be formatted in a more readable way by clnag-format (I don't know why, lol)